packages/jest-schemas/src/raw-types.ts

```diff
@@ -0,0 +1,351 @@
+]);
+
+// TODO: add type test that these are all the colors available in chalk.ForegroundColor
+export const ChalkForegroundColors = Type.Union([
+  Type.Literal('black'),
+  Type.Literal('red'),
+  Type.Literal('green'),
+  Type.Literal('yellow'),
+  Type.Literal('blue'),
+  Type.Literal('magenta'),
+  Type.Literal('cyan'),
+  Type.Literal('white'),
+  Type.Literal('gray'),
+  Type.Literal('grey'),
+  Type.Literal('blackBright'),
+  Type.Literal('redBright'),
+  Type.Literal('greenBright'),
+  Type.Literal('yellowBright'),
+  Type.Literal('blueBright'),
+  Type.Literal('magentaBright'),
+  Type.Literal('cyanBright'),
+  Type.Literal('whiteBright'),
+]);
+
+const RawDisplayName = Type.Object({
+  name: Type.String(),
+  color: ChalkForegroundColors,
+});
+
+// TODO: verify these are the names of istanbulReport.ReportOptions
+export const RawCoverageReporterNames = Type.Union([
+  Type.Literal('clover'),
+  Type.Literal('cobertura'),
+  Type.Literal('html-spa'),
+  Type.Literal('html'),
+  Type.Literal('json'),
+  Type.Literal('json-summary'),
+  Type.Literal('lcov'),
+  Type.Literal('lcovonly'),
+  Type.Literal('none'),
+  Type.Literal('teamcity'),
+  Type.Literal('text'),
+  Type.Literal('text-lcov'),
+  Type.Literal('text-summary'),
+]);
+
+const RawCoverageReporters = Type.Array(
+  Type.Union([
+    RawCoverageReporterNames,
+    Type.Tuple([
+      RawCoverageReporterNames,
+      Type.Record(Type.String(), Type.Unknown()),
+    ]),
+  ]),
+);
+
+const RawGlobalFakeTimersConfig = Type.Partial(
+  Type.Object({
+    enableGlobally: Type.Boolean({
+      description:
+        'Whether fake timers should be enabled globally for all test files.',
+      default: false,
+    }),
+  }),
+);
+
+const RawFakeableAPI = Type.Union([
+  Type.Literal('Date'),
+  Type.Literal('hrtime'),
+  Type.Literal('nextTick'),
+  Type.Literal('performance'),
+  Type.Literal('queueMicrotask'),
+  Type.Literal('requestAnimationFrame'),
+  Type.Literal('cancelAnimationFrame'),
+  Type.Literal('requestIdleCallback'),
+  Type.Literal('cancelIdleCallback'),
+  Type.Literal('setImmediate'),
+  Type.Literal('clearImmediate'),
+  Type.Literal('setInterval'),
+  Type.Literal('clearInterval'),
+  Type.Literal('setTimeout'),
+  Type.Literal('clearTimeout'),
+]);
+
+const RawFakeTimersConfig = Type.Partial(
+  Type.Object({
+    advanceTimers: Type.Union([Type.Boolean(), Type.Number({minimum: 0})], {
+      description:
+        'If set to `true` all timers will be advanced automatically by 20 milliseconds every 20 milliseconds. A custom ' +
+        'time delta may be provided by passing a number.',
+      default: false,
+    }),
+    doNotFake: Type.Array(RawFakeableAPI, {
+      description:
+        'List of names of APIs (e.g. `Date`, `nextTick()`, `setImmediate()`, `setTimeout()`) that should not be faked.' +
+        '\n\nThe default is `[]`, meaning all APIs are faked.',
+      default: [],
+    }),
+    now: Type.Union([Type.Integer({minimum: 0}), Type.Date()], {
```

Comment:
Would recommend removing Type.Date() as this type is a JS extension type in TB (so would be unknown to Ajv). But could represent as the following.

```typescript
Type.Object({
  // ...
  now: Type.Unsafe<number|Date>(Type.Integer({minimum: 0}))
})
```
This would ensure that `now` be integer only, but with the type suggesting a potential for the property to converted into a `Date` object. Have a quick read over the TB [Transform](https://github.com/sinclairzx81/typebox?tab=readme-ov-file#types-transform) feature as this may be able to automatically transform values like this on value load, something that could potentially be run post successful Ajv validate.

Reply:
Yeah, that makes sense!

Reply:
we already do work to make the type different from config input and usage at runtime: https://github.com/jestjs/jest/blob/e20a4be7b91e54eed35842b4799ecec138e82f41/packages/jest-types/src/Config.ts#L96-L102

And yeah, plugging the schema and its defaults with transformers into our [`normalize`](https://github.com/jestjs/jest/blob/e20a4be7b91e54eed35842b4799ecec138e82f41/packages/jest-config/src/normalize.ts#L486) function is a long term plan 🙂